### PR TITLE
Smart, keypoint-aware image cropping augmentation

### PR DESCRIPTION
**Pull request by jeylau**
Image cropping in multi-animal projects is now part of the augmentation pipeline, avoiding cluttered labeled-data folders and simplifying a great number of steps of the DeepLabCut workflow (dataset creation, label checking, network evaluation, etc.).

### CODE REVIEW

File: deeplabcut/pose_cfg.yaml [hunk 1]
```diff
@@ -26,6 +26,16 @@ global_scale: 0.8
...
 batch_size: 1
 
+# Probability with which the augmenters will be applied to input images
+apply_prob: 0.5
+
+# Resize images prior to augmentation
+pre_resize: []  # Specify [width, height] if pre-resizing is desired
+
+# Smart, on-the-fly image cropping, replacing deeplabcut.cropimagesandlabels
+crop_size: [400, 400]  # width, height
```
Comment (MMathisLab) [inline deeplabcut/pose_cfg.yaml:37]: Is `crop_size` also set as a passable list to the create_training_dataset fxns?

Comment (jeylau) [reply deeplabcut/pose_cfg.yaml:37]: Good point! I'll add it :)

**Review by MMathisLab — APPROVED** [review]

**Review by AlexEMG — CHANGES_REQUESTED** [review]
Looks great -- let's make spatial uniform 50% and key point-density 50% the default.

**Review by AlexEMG — APPROVED** [review]
Great stuff, please update the docs and passing of parameters for creating the training data.

**Review by AlexEMG — APPROVED** [review]

**Pull request merged**